### PR TITLE
fix: `drop <table>` for temp tables

### LINKED ISSUE #1855: Unable to drop temp tables
## Description

```
sean@Seans-Air ~ % ./glaredb
GlareDB (v0.5.1)
Using in-memory catalog
Type \help for help.
> create temp table a as select 1;
Table created
> drop table a;
Error: Execution error: failed to drop tables: Failed Metastore request: Missing database object; schema: public, name: a
```

### PR DESCRIPTION
**Pull request by universalmind303**
closes #1855

### CODE REVIEW

File: crates/sqlexec/src/remote/planner.rs [hunk 1]
```diff
@@ -185,12 +192,67 @@ impl ExtensionPlanner for DDLExtensionPlanner {
...
+                        )));
+                    }
+                }
+                let exec: Arc<dyn ExecutionPlan> = match (
+                    temp_table_drops.is_empty(),
+                    drops.is_empty(),
+                ) {
+                    // both temp and remote tables
+                    (false, false) => {
+                        return Err(DataFusionError::Plan("Unable to drop temp and native tables in the same statement. Please use separate statements.".to_string()))?;
```
Comment (universalmind303) [inline crates/sqlexec/src/remote/planner.rs:227]: this does add one new unique constraint that I can't really find a good solution for. 

You now can't drop temp and native tables in the same statement.

File: crates/sqlexec/src/remote/planner.rs [hunk 2]
```diff
@@ -185,12 +192,67 @@ impl ExtensionPlanner for DDLExtensionPlanner {
                 })))
             }
             ExtensionType::DropTables => {
-                let lp = require_downcast_lp::<DropTables>(node);
-                Ok(Some(Arc::new(DropTablesExec {
-                    catalog_version: self.catalog_version,
-                    tbl_references: lp.tbl_references.clone(),
-                    if_exists: lp.if_exists,
-                })))
+                let tmp_catalog = session_state
```
Comment (universalmind303) [inline crates/sqlexec/src/remote/planner.rs:195]: this whole section inspects the catalogs & builds either a `DropTempTablesExec` or a `DropTablesExec`

File: crates/sqlexec/src/remote/planner.rs [hunk 3]
```diff
@@ -306,17 +382,30 @@ impl<'a> RemotePhysicalPlanner<'a> {
...
             for child in plan.children() {
                 match child.as_any().downcast_ref::<RuntimeGroupExec>() {
                     Some(exec) if exec.preference == RuntimePreference::Local => {
+                        if matches!(pref, RuntimePreference::Local) {
+                            // If the root of the plan is local, then we can just
+                            // execute everything locally by omitting the remote
+                            // execution exec.
+                            new_children.push(child);
+                            continue;
+                        }
```
Comment (universalmind303) [inline crates/sqlexec/src/remote/planner.rs:407]: this was needed for some cases where it was trying to push local execs to remote even though the parent was also a local exec. such as `create temp table t1 as select * from generate_series(...)`

Previously this was handled by the rewriter, but now everything happens in a single pass in the extension planner, so some additional logic was needed in here.

**Comment by universalmind303** [comment]
@vrongmeal can you make sure this didn't cause a regression with your `INSERT INTO` PR #1877. I pulled down the changes & tried to resolve the merge conflicts, but want to be 100% sure there wasn't any regression with your PR.

File: crates/sqlexec/src/remote/planner.rs [hunk 4]
```diff
@@ -56,14 +57,18 @@ use crate::planner::physical_plan::show_var::ShowVarExec;
 use crate::planner::physical_plan::update::UpdateExec;
 
 use super::client::RemoteSessionClient;
-use super::rewriter::DDLRewriter;
 
 pub struct DDLExtensionPlanner {
+    catalog: SessionCatalog,
     catalog_version: u64,
```
Comment (vrongmeal) [inline crates/sqlexec/src/remote/planner.rs:63]: Is there a reason we have `catalog_version` stored here? Can't we simply `catalog.version()` where it's required.

File: testdata/sqllogictests/temp_table.slt [hunk 5]
```diff
@@ -108,12 +108,49 @@ select * from t1
...
+
+query II
+select * from tmp1;
+----
+1  2
+
+statement ok
+drop table tmp1;
+
+statement error
```
Comment (vrongmeal) [inline testdata/sqllogictests/temp_table.slt:148]: Let's add the error message here?

**Review by vrongmeal — COMMENTED** [review]
> @vrongmeal can you make sure this didn't cause a regression with your INSERT INTO PR https://github.com/GlareDB/glaredb/pull/1877. I pulled down the changes & tried to resolve the merge conflicts, but want to be 100% sure there wasn't any regression with your PR.

The PR works fine with the `INSERT` changes.

Left a few comments.

**Review by vrongmeal — APPROVED** [review]

**Pull request merged**